service/history/workflow/mutable_state_impl.go

```diff
@@ -4236,6 +4236,51 @@ func (ms *MutableStateImpl) RejectWorkflowExecutionUpdate(_ string, _ *updatepb.
 	return nil
 }
 
+func (ms *MutableStateImpl) AddWorkflowExecutionOptionsUpdatedEvent(
+	versioningOverride *workflowpb.VersioningOverride,
+) (*historypb.HistoryEvent, error) {
+	if err := ms.checkMutability(tag.WorkflowActionWorkflowOptionsUpdated); err != nil {
+		return nil, err
+	}
+	event := ms.hBuilder.AddWorkflowExecutionOptionsUpdatedEvent(versioningOverride)
+	if err := ms.ApplyWorkflowExecutionOptionsUpdatedEvent(event); err != nil {
+		return nil, err
+	}
+	return event, nil
+}
+
+func (ms *MutableStateImpl) ApplyWorkflowExecutionOptionsUpdatedEvent(event *historypb.HistoryEvent) error {
+	override := event.GetWorkflowExecutionOptionsUpdatedEventAttributes().GetVersioningOverride()
+	previousCurrentDeployment := ms.GetEffectiveDeployment()
+	previousEffectiveVersioningBehavior := ms.GetEffectiveVersioningBehavior()
+	ms.GetExecutionInfo().GetVersioningInfo().GetBehavior()
+	ms.GetExecutionInfo().VersioningInfo.VersioningOverride = override
```

Comment:
versioningInfo may be null. User should be add override to unversioned wfs.

Reply:
I'll make it more clear with the variable name, but this is the merged override. So, if the previous override was null and the UpdateOptions request adds a non-null override, then the `VersioningOverride` stored in the history event will be the new non-null override. That's what we are applying here.

I can rename the `override` variable to `mergedOverride` or `newOverride` if that makes it more clear

Reply:
I mean what if ms.GetExecutionInfo().VersioningInfo itself is null? how can you set ms.GetExecutionInfo().VersioningInfo.VersioningOverride then?

Reply:
yeah, you need to check if it is `nil` or not, and if `nil` create is first.

Reply:
ohh yes thank you!

Reply:
done :)